ui/raidboss/data/07-dt/eureka/occult_crescent_south_horn.ts

```diff
@@ -76,6 +113,44 @@ const triggerSet: TriggerSet<Data> = {
           console.log(`Start CE: ??? (${ceId})`);
       },
     },
+    {
+      id: 'Occult Crescent Phantom Job Tracker',
+      // data0 also contains a Phantom Job id and level
+      // Seems to follow this pattern XYZ, where X is Job #, Y is always 0, and Z is the level
+      // First digit is the job:
+      // Thief = C
+      // Oracle = B
+      // Chemist = A
+      // Cannoneer = 9
+      // Time Mage = 8
+      // Geomancer = 7
+      // Bard = 6
+      // Samurai = 5
+      // Ranger = 4
+      // Monk = 3
+      // Berserker = 2
+      // Knight = 1
+      // Freelancer = null
+      // Freelancer level is accumulation of maxed jobs +1 and seems like only 3 more jobs, ending at F
+      type: 'GainsEffect',
+      netRegex: { effectId: [...phantomJobEffectIds], capture: true },
+      condition: Conditions.targetIsYou(),
+      run: (data, matches) => {
+        data.phantomJob = matches.effectId;
+        const jobData = matches.data0;
+
+        // Assuming this isn't possible given the filter on statuses
+        if (jobData === undefined)
+          return;
+
+        // Get Phantom Job Level
+        // Freelancer has a length 2 data0, whereas the others have length 3
+        // Remove first element in the data0 string for other jobs
+        data.phantomJobLevel = (data.phantomJob === phantomJobData.freelancer)
+          ? parseInt(jobData, 16)
+          : parseInt(jobData.slice(1), 16);
```

Comment:
`data0` has some strange padding going on, but it's really two `byte` values. The more "correct" way to handle this, IMO, is to left pad to 4 characters and treat it as two bytes, rather than just relying on there not being more than 3 new jobs in the next instance.

Something like this:

```suggestion
        const jobData = matches.data0?.padStart(4, '0');

        // Assuming this isn't possible given the filter on statuses
        if (jobData === undefined)
          return;

        data.phantomJobLevel = parseInt(jobData.slice(2), 16);
```

Reply:
Ah, that padding to ensure it's always 4 makes sense.